client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -106,6 +128,37 @@ tie.factory('PythonCodeRunnerService', [
       });
     };
 
+    // Converts the output returned from running the code.
+    // Before: array in which each entry was one line of stdOut
+    // After: array in which the i-th entry is the string to be displayed for
+    // that i-th test case
+    var _standardizeOutputInClient = function(stdOutArray, separator) {
+      if (stdOutArray.length === 0) {
+        return [];
+      }
+      var standardizedOutput = [];
+      var currentStringStartIndex = 0;
+      var currentStringEndIndex = 0;
+      for (currentStringEndIndex; currentStringEndIndex < stdOutArray.length;
+        currentStringEndIndex++) {
+        if (stdOutArray[currentStringEndIndex] === separator) {
+          // Convert the sub-array into a condensed string of the entire
+          // stdOut for that given test case.
+          var currentStdOutStringRepresentation = '';
+          for (var i = currentStringStartIndex;
```

Comment:
This seems very manual. What about the following approach:

- Concatenate stdOut into a string (which it might already be originally?)
- Split the string by the separator value to get an array for each test case